### ISSUE TITLE
Add support of templates in separated html files

### ISSUE THREAD

**Issue by code81** [vuejs/eslint-plugin-vue#605]
* **ESLint Version:** 4.16.0
* **eslint-plugin-vue Version:** 4.7.1
* **Node Version:** 9.4.0

**Full configuration:**
```
{
  "extends": [
    "standard",
    "plugin:vue/recommended"
  ],

  "parser": "vue-eslint-parser",

  "parserOptions": {
    "sourceType": "module",
    "parser": "babel-eslint",
    "newline-before-return": true,
    "allowImportExportEverywhere": true
  },

  "env": {
    "jasmine": true
  },

  "rules": {
    "vue/comment-directive": "off"
  }
}
```

**What did you do?**
package.json:
   "lint": "eslint ./src/**/*.{js,html}"

Run "npm run lint".

```
compA/index.html:
<div id="compA" title="compATitle">
  {{ content }}
</div>

compA/index.js:
import template from './index.html'

// @vue/component
export default {
  props: {
    content: {
      type: String,
      required: true
    }
  },

  template
}
```

**What did you expect to happen?**
Some lint errors from template.

**What actually happened?**
No lint errors from template.

When I add root tag &lt;template&gt; to the template, lint errors are present in output, but Vue is complaining: Cannot use &lt;template&gt; as component root element because it may contain multiple nodes.
